osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/factories/InvocationCounterManager.java

```diff
@@ -59,21 +59,32 @@ public InvocationCounterManager(
    */
   public void initializeInvocationCounter(
       final MessageMetadata messageMetadata, final DlmsDevice device) throws OsgpException {
+    /*
+     * When the invocation counter is out of sync, the device closes the session.
+     * By setting the ip-address to null, the application will be forced to get a new ip-address.
+     * The meter will start a new session.
+     * This is done by the DlmsConnectionFactory in the method:
+     * this.domainHelperService.setIpAddressFromMessageMetadataOrSessionProvider
+     */
+    device.setIpAddress(null);
 
     this.initializeWithInvocationCounterStoredOnDevice(messageMetadata, device, null);
+
+    /*
+     * When the invocation counter has been initialized, the device closes the session.
+     * By setting the ip-address to null, the application will be forced to get a new ip-address.
```

Comment:
This comment is the same as the previous comment. To make it clear that it's really the intention to do this twice, I suggest to change it to: "by setting the ip-address to null again".

Reply:
Changed